public/controlpanel.php

```diff
@@ -148,6 +159,17 @@ function confirmEmailChange(event) {
                     echo "</td>";
                     echo "</tr>";
                 }
+                echo "<tr>";
+                echo "<td>Always expand User Awards on profiles</td>";
+                echo "<td>";
+                echo "<div class='flex flex-col'>";
+                echo "<div>";
+                echo "<input onclick='handleExpandedAwardsCheckboxClick(event)' class='mr-2' type='checkbox' " . ($doesPreferAlwaysExpandedRewards ? 'checked' : '') . ">";
+                echo "<button class='btn mr-4' onclick='expandableAwards.saveExpandableAwardsPreference(\"$expandedAwardsCookieName\", isExpandedAwardsCheckboxChecked)'>Save</button>";
+                echo "</div>";
+                echo "<span class='text-xs'>This only applies to your current device.</span>";
+                echo "</div>";
+                echo "</td>";
```

Comment:
I'd prefer there to be an implicit way of remembering this setting.
Instead of adding a setting to the controlpanel, there could be a 'Collapse' button that resets the preference. No need for a use setting here imo.

Regardless of that, if visual feedback is given it should re-use existing features, I removed a lot of alerts in the past, confirm dialogs will be replaced with dialogs in the future.
![Screenshot 2023-03-02 at 19 58 13](https://user-images.githubusercontent.com/1280590/222527303-618aaef0-10ae-45a9-b64d-b54d9207fe97.png)

Reply:
Your proposed approach definitely seems better. I was unclear how to get the "Changed" message to appear, so I naively went with `alert()` 😈

I will move this persistence entirely to the expand button itself and add make it more like a toggle-button, allowing the user to collapse the list after expansion.

Reply:
I've reverted all changes on _controlpanel.php_. Per your suggestion, this is now totally persisted by the button itself. When clicking the button to Expand, rather than it being removed from the DOM, it becomes a "Collapse" button. Clicking it again adjusts the persisted setting to the user's preference.

I've updated the tests to ensure this behavior is working correctly.